### PR TITLE
test: add Compose UI tests for desktop-core dashboards

### PR DESCRIPTION
**Pull request by kaeawc**
## Summary
- Add Compose UI tests for `NavigationDashboard` (FlowMapListView, ScreenDetailView, TransitionDetailView, StatItem, DetailRow)
- Add Compose UI tests for `LayoutInspectorDashboard` (PropertyInspectorPanel, BreadcrumbBar)
- Add Compose UI tests for `TestDashboard` (TestDashboardHome loading, error, empty, and data states)
- Fix `Result.Error` constructor calls to use `RuntimeException` wrapper in existing tests
- Make `TestDashboardHome` internal for test accessibility

## Test plan
- [x] All new UI tests pass via `./gradlew :desktop-core:test`
- [x] Existing tests still pass (no regressions)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: android/desktop-core/src/test/kotlin/dev/jasonpearson/automobile/desktop/core/navigation/NavigationDashboardUiTest.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        setContent {
+            MaterialTheme {
+                TransitionDetailView(
+                    transition = noLatency,
+                    onBack = {},
+                    onScreenSelected = {},
+                )
+            }
+        }
+        onNodeWithText("Avg Latency").assertDoesNotExist()
```
[the remark this answers is not in the document]
Comment (kaeawc) [reply android/desktop-core/src/test/kotlin/dev/jasonpearson/automobile/desktop/core/navigation/NavigationDashboardUiTest.kt]: Same as above — `assertDoesNotExist()` is a member of `SemanticsNodeInteraction`, no separate import needed. Tests compile and pass.
